### DISCUSSION TITLE
Analog input gain - is it possible to go over +6 dB?

### DISCUSSION THREAD

**Discussion by pszklarski** [geoffreybennett/alsa-scarlett-gui#96] [Q&A] [answered]
Hi! First of all - thank you for creating such great software :)
Question: is it possible to add more gain to the analog inputs in the mixer (over +6 dB) or is it a hardware limitation (Scarlett 8i6 3rd gen)?

**Comment by geoffreybennett** [comment]
Hi @pszklarski,

You're welcome, thank you!

That's a very interesting question! My implementation of a +6dB limit was to match the Focusrite Control software (it was among the first things I implemented in the Gen 2 driver, and it's used identically in every series since). If you check the driver code, the `scarlett2_mixer_values[]` array translates from dB values ("`(dB + 80) * 2`") to the 16-bit linear value which gets uploaded to the device.

+6dB corresponds to 16345 (8192 is treated by the device as unity gain). Do you want to try increasing those values to see what happens? 16345 is only a 14-bit value, so perhaps you can get +12dB gain with a mixer value of 32612 (if the firmware treats it as a signed value) or even +18dB (65071, if it's an unsigned value). Figuring this out will require some experimentation, which I don't have time for right now, but if you can do it, I'd be interested in the results.

Another (kludge) possibility that doesn't need any driver modifications: if you are not using all the Mixer Inputs you can assign the same channel to two Mixer Inputs and then you can double the volume, on top of the 6dB gain, like this:

![image](https://github.com/geoffreybennett/alsa-scarlett-gui/assets/1866657/2632d5c6-6d61-4c76-9cf1-5d9222cb6c7c)

(you can also go crazy and send Mixer Outputs back to Mixer Inputs and amplify further the already-amplified signal)

Regards,
Geoffrey.

**Comment by geoffreybennett** [comment, marked as answer]
Hi again @pszklarski,

I did some experimentation this evening and found that it is no problem to increase the mixer controls' range up to +12dB. Up to +18dB also initially works, but on some interfaces the value is treated as a signed value when it's read back at startup from the saved configuration. So +18dB which is 65071 internally gets read back as −465 which is −25dB and because it's negative the phase is inverted:

![image](https://github.com/geoffreybennett/alsa-scarlett-gui/assets/1866657/653d5747-7eb4-4111-8929-27d21d0a0acd)

I'd call the signed/unsigned confusion a firmware bug, and I'd also suggest that the chances of it being fixed are really low :). So I will submit a patch upstream to increase the mixer range up to +12dB:

![image](https://github.com/geoffreybennett/alsa-scarlett-gui/assets/1866657/a0f04dd6-bdd2-4e39-82ed-e795c6f62509)

Regards,
Geoffrey.

**Comment by geoffreybennett** [comment]
Submitted upstream: https://lore.kernel.org/linux-sound/Zj+gYT4F2XeKTD93@m.b4.vu/
In my repo: https://github.com/geoffreybennett/scarlett-gen2/commit/f8cc36c8f1601ad305ab82966959077d7144389e
And: https://github.com/geoffreybennett/scarlett-gen2/releases
